### PR TITLE
[HttpFoundation] Use UPSERT for sessions stored in PgSql >= 9.5

### PR DESCRIPTION
**Pull request by nicolas-grekas**
| Q | A |
| --- | --- |
| Branch? | 2.7 |
| Bug fix? | yes |
| New feature? | no |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets | - |
| License | MIT |
| Doc PR | - |

### CODE REVIEW

File: src/Symfony/Bridge/Doctrine/HttpFoundation/DbalSessionHandler.php [hunk 1]
```diff
@@ -224,11 +224,11 @@ private function getMergeSql()
     {
         $platform = $this->con->getDatabasePlatform()->getName();
 
-        switch ($platform) {
-            case 'mysql':
+        switch (true) {
```
Comment (nicolas-grekas) [inline src/Symfony/Bridge/Doctrine/HttpFoundation/DbalSessionHandler.php]: bonus fix: https://github.com/symfony/symfony/pull/19048/files#diff-e6a5cc5d66fa241233b3314fcd49f228R236 is dead code today

File: src/Symfony/Component/HttpFoundation/Session/Storage/Handler/PdoSessionHandler.php [hunk 2]
```diff
@@ -659,11 +659,11 @@ private function getSelectSql()
      */
     private function getMergeSql()
     {
-        switch ($this->driver) {
-            case 'mysql':
+        switch (true) {
```
Comment (nicolas-grekas) [inline src/Symfony/Component/HttpFoundation/Session/Storage/Handler/PdoSessionHandler.php]: https://github.com/symfony/symfony/pull/19048/files#diff-5b96aee9c7c07a0e18d1fd32f7a3028eR671 dead code today

**Comment by fabpot** [comment]
👍

**Comment by Tobion** [comment]
Please update https://github.com/nicolas-grekas/symfony/blob/0b5b693f19915a263630ea85bc3cce208f9bad9d/src/Symfony/Component/HttpFoundation/Session/Storage/Handler/PdoSessionHandler.php#L350 to say "like in Postgres < 9.5" to avoid confusion.

Apart from that :+1:

**Comment by nicolas-grekas** [comment]
updated

**Comment by fabpot** [comment]
Thank you @nicolas-grekas.

**Pull request merged**